instrumentation/apache-httpclient/apache-httpclient-5.0/library/src/test/java/io/opentelemetry/instrumentation/apachehttpclient/v5_0/ApacheHttpClient5Test.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.apachehttpclient.v5_0;
+
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.AbstractHttpClientTest;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpClientInstrumentationExtension;
+import java.util.Collections;
+import org.apache.hc.client5.http.config.ConnectionConfig;
+import org.apache.hc.client5.http.config.RequestConfig;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.HttpClientBuilder;
+import org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager;
+import org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManagerBuilder;
+import org.apache.hc.core5.util.Timeout;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class ApacheHttpClient5Test extends AbstractApacheHttpClient5Test {
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = HttpClientInstrumentationExtension.forLibrary();
+
+  @Override
+  protected InstrumentationExtension testing() {
+    return testing;
+  }
+
+  @Override
+  protected CloseableHttpClient createClient(boolean readTimeout) {
+    HttpClientBuilder builder =
+        ApacheHttpClient5Telemetry.builder(testing.getOpenTelemetry())
+            .setCapturedRequestHeaders(
+                Collections.singletonList(AbstractHttpClientTest.TEST_REQUEST_HEADER))
+            .setCapturedResponseHeaders(
+                Collections.singletonList(AbstractHttpClientTest.TEST_RESPONSE_HEADER))
+            .build()
+            .newHttpClientBuilder();
+    builder.setDefaultRequestConfig(
+        RequestConfig.custom()
+            .setCircularRedirectsAllowed(true)
+            .setMaxRedirects(2)
```

Comment:
I tried without it as well, and it still does not work 😢

Reply:
did you try changing the max redirects to 1?

Reply:
If I do that it still fails and the test `io.opentelemetry.instrumentation.testing.junit.http.AbstractHttpClientTest#basicRequestWith2Redirects` fails as well

Reply:
I had a look at this again, it seems to be failing for a different reason from what I remember from looking at it first, Idk maybe I messed up something. Anyway the reason seems to be that the code you copy pasted from the earlier instrumentation that is supposed to detect circular redirects doesn't work for this version. I think it expects to get an exception in case of circular redirect so that the span could be ended in https://github.com/elbiocaetano/opentelemetry-java-instrumentation/blob/d55d1db64f9cedb88eb19bb0e69705335d31b534/instrumentation/apache-httpclient/apache-httpclient-5.2/library/src/main/java/io/opentelemetry/instrumentation/apachehttpclient/v5_2/OtelExecChainHandler.java#L143 but in this version calling `getLocationURI` does not throw a `CircularRedirectException`.
OpenTelemetry spec (https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#http-client) describes two kinds of http client instrumentations, the first kind creates a span for each redirect (we call it `low level`) and the second kind creates a span that encompasses all the redirect attempts (we call it `high level`). The current recommended approach would be the `low level`, but historically the spec only described the `high level` one. The instrumentation you built uses the `high level` approach but I suspect that here the `low level` one would be easier, you wouldn't then really need to deal with the circular redirects. In tests you can enable the `low level` with https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/e4445ab507d8c0c76289df7ab7c32acb9d627547/instrumentation/okhttp/okhttp-3.0/testing/src/main/java/io/opentelemetry/instrumentation/okhttp/v3_0/AbstractOkHttp3Test.java#L116

Reply:
I change apache http client 4.3 instrumentation to `low level` in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10253

Reply:
Thanks for the tip! With that I managed to solve the problem. Only changing to low level instrumentation was not enough. Instead I implemented the circular redirect validation that is present in 4.3 and now we have the expected behaviour. Could you take a look if the solution is valid?

Reply:
@laurit , could you take another look, please?